core/chain-configs/src/client_config.rs

```diff
@@ -801,6 +801,8 @@ pub struct ClientConfig {
     pub save_trie_changes: bool,
     /// Whether to persist transaction outcomes to disk or not.
     pub save_tx_outcomes: bool,
+    /// Whether to persist state changes on disk or not.
+    pub save_state_changes: bool,
```

Comment:
I would rather our configuration be in terms of functional components, not behavioural ones. Alas the ship seems like it might have sailed. But maybe we can start now?

To clarify what I mean by "functional": "enable_rosetta" versus "do_or_not_do_something_rosetta_needs_to_operate". In this case it should be enough to toggle saving of state changes, but in the future there may be some functions where multiple of them depend on the same behaviour at the same time (and so avoiding that behaviour would require them all to be turned off.) "enable/disable rosetta" has also a much more comprehensible effect to the end user without additional documentation.